platform/darwin/src/MGLGeometry.h

```diff
@@ -45,6 +45,23 @@ typedef struct MGLCoordinateBounds {
     CLLocationCoordinate2D ne;
 } MGLCoordinateBounds;
 
+/** 
+ A quadrilateral area as measured on a two-dimensional map projection.
+ `MGLCoordinateQuad` differs from `MGLCoordinateBounds` in that it allows
+ representation of non-axis aligned bounds and non-rectangular quadrilaterals
+ */
+typedef struct MGLCoordinateQuad {
+    /** Coordinate at the top left corner. */
+    CLLocationCoordinate2D topLeft;
+    /** Coordinate at the top right corner. */
+    CLLocationCoordinate2D topRight;
+    /** Coordinate at the bottom right corner. */
+    CLLocationCoordinate2D bottomRight;
+    /** Coordinate at the bottom left corner. */
+    CLLocationCoordinate2D bottomLeft;
```

Comment:
The reason I happened to be aware of `NSEdgeInset`’s ordering is that it’s opposite that of the style specification, which is [clockwise](https://www.mapbox.com/mapbox-gl-js/style-spec/#layout-symbol-icon-text-fit-padding) (top, right, bottom, left).

If this struct ends up differing from the style specification’s [clockwise](https://www.mapbox.com/mapbox-gl-js/style-spec/#sources-image) order, we should note the difference in [this guide](https://github.com/mapbox/mapbox-gl-native/blob/5e0018cdc3844994bf4afdc0a5e6ae90636c9967/platform/darwin/docs/guides/For%20Style%20Authors.md.ejs#L287-L296).